sos/report/plugins/mellanox_firmware.py

```diff
@@ -1,132 +1,481 @@
+                ]
+            )
+
+        # Add additional commands only if firmware is not secure
+        if not self.is_secured_fw():
+            self._commands[2].extend(
+                [
+                    f"flint -d {self.device} dc",
+                ]
+            )
+
+        return self._commands
+
+
+class MSTFlintFirmwareTool(BaseFirmwareTool):
+    """
+    Firmware tool implementation for Mellanox MSTFlint utilities.
+    """
+
+    def is_secured_fw(self):
+        """
+        Check if the firmware is secure using the mstflint query command.
+
+        Returns:
+            Boolean indicating secure firmware status.
+        """
+        result = self.plugin.exec_cmd(
+            f"mstflint -d {self.device} q full",
+        )
+
+        attrs = self.parse_security_attributes(
+            result.get("output", "")
+        ) if result.get("status") == 0 else []
+
+        return "secure-fw" in attrs and "dev" not in attrs
+
+    @property
+    def commands(self):
+        """
+        Build and return all commands for MSTFlint firmware collection.
+
+        Commands are grouped by priority:
+          - 0: High-priority queries and repeated debug commands.
+          - 1: Medium-priority debug commands.
+          - 2: Extended diagnostic commands, global commands, and
+               optional commands for unsecured firmware.
+
+        Returns:
+            Dictionary mapping priority -> list of commands.
+        """
+        if self._commands is not None:
+            return self._commands
+
+        self._commands = {
+            0: [],
+            1: [],
+            2: [],
+        }
 
+        self._commands[0].extend(
+            [
+                f"mstflint -d {self.device} q full",
+                # 'mstregdump' repeated intentionally for diagnostic purposes
+                *[f"mstregdump {self.device}" for _ in range(3)]
+            ]
+        )
+
+        self._commands[1].extend(
+            [
+                f"mstresourcedump dump -d {self.device} "
+                f"--segment BASIC_DEBUG"
+            ]
+        )
+
+        self._commands[2].extend(
+            [
+                f"mstconfig -d {self.device} -e q",
+                f"mstreg -d {self.device} --reg_name ROCE_ACCL --get",
+                f"mstlink -d {self.device}",
+            ]
+        )
+
+        # Global commands should only run once (on the first/primary device)
+        if self.primary:
+            self._commands[2].extend(
+                [
+                    "mstflint --version",
+                    "mstdevices_info"
+                ]
+            )
+
+        # Add additional commands only if firmware is not secure
+        if not self.is_secured_fw():
+            self._commands[2].extend(
+                [
+                    f"mstflint -d {self.device} dc"
+                ]
+            )
+
+        return self._commands
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
```

Comment:
Lets change the order, plugin class should be at the top, it makes the code look clearer